src/PositionManager.sol

```diff
@@ -202,16 +212,28 @@ contract PositionManager is
         liquidity = poolManager.getPositionLiquidity(config.poolKey.toId(), positionId);
     }
 
-    /// @dev Send excess native tokens back to the recipient (locker)
-    /// @param recipient the receiver of the excess native tokens. Should be the caller, the one that sent the native tokens
-    function _sweepNativeToken(address recipient) internal {
-        uint256 nativeBalance = address(this).balance;
-        if (nativeBalance > 0) recipient.safeTransferETH(nativeBalance);
+    /// @notice Sweeps the entire contract balance of specified currency to the recipient
+    /// @param params an encoding of Currency, address
+    function _sweep(bytes calldata params) internal {
+        (Currency currency, address to) = abi.decode(params, (Currency, address));
+        uint256 balance = currency.balanceOfSelf();
+        if (balance > 0) currency.transfer(to, balance);
     }
 
     // implementation of abstract function DeltaResolver._pay
-    function _pay(Currency token, address payer, uint256 amount) internal override {
-        // TODO: Should we also support direct transfer?
-        permit2.transferFrom(payer, address(poolManager), uint160(amount), Currency.unwrap(token));
+    function _pay(Currency currency, address payer, uint256 amount) internal override {
+        if (payer == address(this)) {
+            // TODO: currency is guaranteed to not be eth so the native check in transfer is not optimal.
+            currency.transfer(address(poolManager), amount);
+        } else {
+            permit2.transferFrom(payer, address(poolManager), uint160(amount), Currency.unwrap(currency));
+        }
+    }
+
+    function _getFullSettleAmount(Currency currency) private view returns (uint256 amount) {
```

Comment:
how do you feel about putting this in DeltaResolver? 👀 i also have this exact code lol

Reply:
sure!